libs/pilight/protocols/433.92/kambrook.c

```diff
@@ -0,0 +1,228 @@
+	kambrook->message = json_mkobject();
+	json_append_member(kambrook->message, "id", json_mknumber(id, 0));
+	json_append_member(kambrook->message, "unit", json_mknumber(unit, 0));
+	if(state == 0) {
+		json_append_member(kambrook->message, "state", json_mkstring("on"));
+	} else {
+		json_append_member(kambrook->message, "state", json_mkstring("off"));
+	}
+}
+
+
+static void createLow(int s, int e) {
+	int i;
+
+	for(i=s;i<=e;i+=2) {
+		kambrook->raw[i]=(AVG_PULSE_LENGTH);
+		kambrook->raw[i+1]=(AVG_PULSE_LENGTH);
+	}
+
+}
+
+static void createHigh(int s, int e) {
+	int i;
+
+	for(i=s;i<=e;i+=2) {
+		kambrook->raw[i]=(PULSE_MULTIPLIER*AVG_PULSE_LENGTH);
+		kambrook->raw[i+1]=(AVG_PULSE_LENGTH);
+	}
+}
+static void clearCode(void) {
+	createLow(0,94);
+}
+
+static void createHeader(void){
+        int i;
+        for(i=2;i<16;i+=4) {
+                createHigh(i,i+1);
+        }
+}
+
+static void createId(int id) {
+        int binary[255];
+        int length = 0;
+        int i=0, x=0;
+
+        length = decToBinRev(id, binary);
+	for(i=0;i<=length;i++) {
+                if(binary[i]==1) {
+                        x=i*2;
+                        createHigh(62-x,62-x+1);
+                }
+        }
+}
+
+static void createFooter(void) {
+        createHigh(80,94);
+        kambrook->raw[95]=(PULSE_DIV*AVG_PULSE_LENGTH);
+}
+
+static void createOverallCode(int unit, int state) {
+        int binary[255];
+        int length = 0;
+        int i=0, x=0;
+	int overallcode=0;
+
+	while(unit>5) {
+		overallcode+=16;
+		unit-=5;
+	}
+	overallcode+=(unit*2)-1+state;
+
+        length = decToBinRev(overallcode, binary);
+        for(i=0;i<=length;i++) {
+                if(binary[i]==1) {
+                        x=i*2;
+                        createHigh(78-x, 78-x+1);
+                }
+        }
+}
+
+
+
+static int createCode(struct JsonNode *code) {
+	int id = -1;
+	int unit = -1;
+	int state = -1;
+	double itmp = 0;
+
+	if(json_find_number(code, "id", &itmp) == 0)
+		id = (int)round(itmp);
+	if(json_find_number(code, "unit", &itmp) == 0)
+		unit = (int)round(itmp);
+	if(json_find_number(code, "off", &itmp) == 0)
+		state=1;
+	else if(json_find_number(code, "on", &itmp) == 0)
+		state=0;
+
+	if(id == -1 || unit == -1 || state == -1) {
+		logprintf(LOG_ERR, "kambrook: insufficient number of arguments");
+		return EXIT_FAILURE;
+	} else if(id > 29999 || id < 1) {
+		logprintf(LOG_ERR, "kambrook: invalid id range");
+		return EXIT_FAILURE;
+	} else if(unit > 19 || unit < 1) {
+		logprintf(LOG_ERR, "kambrook: invalid unit range");
+		return EXIT_FAILURE;
+	} else {
+		createMessage(id, unit, state);
+		clearCode();
+        createHeader();
+        createId(id);
+        createOverallCode(unit,state);
+        createFooter();
+		kambrook->rawlen = RAW_LENGTH;
+	}
+	return EXIT_SUCCESS;
+}
+
+static void printHelp(void) {
+        printf("\t -i --id=id\tcontrol a device with this id (id) (1-29999)\n");
```

Comment:
`(id) (1-29999)`
You can just add this information to the wiki. None of the protocols have it. Same counts for the unit.